### ISSUE TITLE
CMC set I3Particle::shape to  I3Particle::CascadeSegment (Trac #1787)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1779]
From ticket #1755. jvs:
cmc doesn't do anything to cascades with energies lower than 1 TeV, so we still need to use cascade extension in general. I propose a variant of Claudio's suggestion:
Add a new particle shape, I3Particle::CascadeSegment
When cmc breaks up a cascade, it should emit particles with shape CascadeSegment and set their length to the distance between particles (3 radiation lengths by default)
clsim should simulate the longitudinal distribution of particles with shape Cascade as it currently does. For particles with shape CascadeSegment, it should distribute steps uniformly along the length of the segment.
When cmc simulates the cascade directly to account for the LPM effect, it still bins the results in radiation lengths (see I3CascadeSimulation::Deposit(Particle particle)), so the same treatment is valid no matter which mode is used.

Migrated from https://code.icecube.wisc.edu/ticket/1787
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:35", 
    "description": "From ticket #1763. jvs:\ncmc doesn't do anything to cascades with energies lower than 1 TeV, so we still need to use cascade extension in general. I propose a variant of Claudio's suggestion:\nAdd a new particle shape, I3Particle::CascadeSegment\nWhen cmc breaks up a cascade, it should emit particles with shape CascadeSegment and set their length to the distance between particles (3 radiation lengths by default)\nclsim should simulate the longitudinal distribution of particles with shape Cascade as it currently does. For particles with shape CascadeSegment, it should distribute steps uniformly along the length of the segment.\nWhen cmc simulates the cascade directly to account for the LPM effect, it still bins the results in radiation lengths (see I3CascadeSimulation::Deposit(Particle particle)), so the same treatment is valid no matter which mode is used.", 
    "reporter": "juancarlos", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067215093672", 
    "component": "combo simulation", 
    "summary": "CMC set I3Particle::shape to  I3Particle::CascadeSegment", 
    "priority": "major", 
    "keywords": "cmc", 
    "time": "2016-07-22T14:50:39", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "enhancement"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160722T16:22:48: 
* olivas changed owner from "" to "syoshida"
* olivas changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20161003T13:07:52: 
* jvansanten changed status from "assigned" to "accepted"
* jvansanten changed owner from "syoshida" to "jvansanten"

**Comment by icecube-trac** [comment]
Trac update at 20161003T14:47:27: 
* jvansanten changed resolution from "" to "fixed"
* jvansanten commented:

> In changeset 150460:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="150460"
> Mark sub-cascades as segments. Fixes #1779.
> 
> Along the way, clip some methods that have been unused since PSInterface
> days.
> ```


* jvansanten changed status from "accepted" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:35: 
* olivas commented:

> Milestone IceSim 5.1 deleted


* olivas changed milestone from "IceSim 5.1" to ""
